discordBot.js

```diff
@@ -1,57 +1,110 @@
-    const client = this.client;
-    client.once('ready', () => {
-      console.log('Discord Ready!');
-    });
-
-    client.on('guildMemberAdd', member => {
-      const channel = member.guild.channels.cache.find(ch => ch.name === 'member-log');
-
-      if (!channel) return;
-      channel.send(`Welcome to the server, ${member}`);
-    });
-
-    client.on('message', async message => {
-      const PREFIX = '*';
-      if (message.content.startsWith(PREFIX)) {
-        const input = message.content.slice(PREFIX.length).trim().split(' ');
-        const command = input.shift();
-        const commandArgs = input.join(' ');
-
-        if (command === 'pm') {
-          //TODO parse the argument
-          message.channel.send(`send a private message: ${commandArgs}`);
-        }
-
-        if (command === 'gm') {
-          message.channel.send(`send a group message: ${commandArgs}`);
-        }
-
-        if (command === 'm') {
-          const usernames = await this.getMembers().map(m => m.username);
-          message.channel.send(`Here are all the members: ${usernames}`);
-        }
-      }
-    });
-
-     return client.login(token);
+ start() {
+    try {
+      const client = this.client;
+      const loginStatus = client.login(this.token);
+      client.once('ready', () => {
+        console.log('Discord Ready!');
+      });
+      return loginStatus;
+    } catch(err){
+      return err;
+    }
+
+    // need to be able to give the ability to the programmer to make commands 
+    // without interacting directly with discord api if we want to add that functionality
+
+    // client.on('guildMemberAdd', member => {
+    //   const channel = member.guild.channels.cache.find(ch => ch.name === 'member-log');
+
+    //   if (!channel) return;
+    //   channel.send(`Welcome to the server, ${member}`);
+    // });
+
+    // client.on('message', async message => {
+    //   const PREFIX = '*';
+    //   if (message.content.startsWith(PREFIX)) {
+    //     const input = message.content.slice(PREFIX.length).trim().split(' ');
+    //     const command = input.shift();
+    //     const commandArgs = input.join(' ');
+
+    //     if (command === 'pm') {
+    //       //TODO parse the argument
+    //       message.channel.send(`send a private message: ${commandArgs}`);
+    //     }
+
+    //     if (command === 'gm') {
+    //       message.channel.send(`send a group message: ${commandArgs}`);
+    //     }
+
+    //     if (command === 'm') {
+    //       const usernames = await this.getMembers().map(m => m.username);
+    //       message.channel.send(`Here are all the members: ${usernames}`);
+    //     }
+    //   }
+    // });
   }
 
+  //need to be able to send to any channel or add another function to send to specific 
+  //channels so must take channel id as an argument
   sendMessageToAllChannels(msg) {
-    const sendAll = this.client.channels.cache
+    this.client.on('ready', () => {
```

Comment:
the surrounding
`this.client.on('ready', () => { .. }`
could be omitted

Reply:
sometimes it doesn't work for me if start doesn't finish before this runs. using on ready assures that none of the functions runs before start finishes and we can send messages. let me know if there is another possible solution.

Reply:
I updated the `/test/discordBot.test.js` in the https://github.com/CSCI-49900-Fall-2020/Project-chatmanagerlibrary/pull/7 so it should be an issue, you could have before hook in `mocha` and put done in the callback function, so the unit test would wait the `done` before start the other unit test